### ISSUE TITLE
Config field backend model (system.xml) not used for default value (config.xml)

### ISSUE THREAD

**Issue by kweij** [magento/magento2#7741]
### Preconditions
Magento 2.1.2

### Steps to reproduce
Create a config field with a custom backend model, such as:

```xml
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
    <system>
        <tab id="oops" sortOrder="110">
            <label>Oops</label>
        </tab>
        <section id="oops_theme" type="text" sortOrder="10" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>Theme</label>
            <tab>oops</tab>
            <resource>Oops_Theme::config</resource>
            <group id="homepage" type="text" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1">
                <label>Homepage</label>
                <field id="menu" type="text" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
                    <label>Primary Actions</label>
                    <frontend_model>Oops\Theme\Block\Adminhtml\Form\Field\Menu</frontend_model>
                    <backend_model>Oops\Theme\Model\System\Config\Backend\Json</backend_model>
                </field>
            </group>
        </section>
    </system>
</config>
```

Here I use backend model ```Oops\Theme\Model\System\Config\Backend\Json```. This model is looks like this:

```php
namespace Oops\Theme\Model\System\Config\Backend;

use Magento\Framework\App\Config\Value;

class Json extends Value
{
    /**
     * Process data after load
     *
     * @return void
     */
    protected function _afterLoad()
    {
        $value = $this->getValue();
        $value = json_decode($value, true);
        $this->setValue($value);
    }

    /**
     * Prepare data before save
     *
     * @return void
     */
    public function beforeSave()
    {
        $value = $this->getValue();
        unset($value['__empty']);
        $value = json_encode($value, JSON_PRETTY_PRINT);
        $this->setValue($value);
    }
}
```

### Expected result
Performs ```json_decode()``` after loading data from the configuration and ```json_encode()``` before saving the data to the database. **Expected** is that the ```_afterLoad()``` method gets evoked, unregardless of wether the configuration is retrieved from database or ```config.xml```, however ...

### Actual result
The ```_afterLoad()``` method gets evoked _only_ after loading the configuration value from the database, and _not_ after loading the configuration from ```config.xml```. This causes default configuration from ```config.xml``` to be passed to the form-element as ```string``` in stead of ```Array```, resulting in empty configuration fields in the admin interface.

**Comment by kweij** [comment]
This can be fixed easily, as in pull request #7742

**Comment by kweij** [comment]
Ok, well since my pull-request won't build due to _unexpected_ results (it's not crashing, but it alters the way default values in ``config.xml`` are treated, so it's not backwards compatible) this bug/change may require a different approach. Nevertheless, I stand by my expected result in the original post.

**Comment by okorshenko** [comment]
Here is a ticket MAGETWO-70797 to track the issue progress

**Comment by RomaKis** [comment]
@okorshenko, @ishakhsuvarov, please, close this ticket as the [pull request](https://github.com/magento/magento2/issues/7741) was successfully merged.

**Comment by ishakhsuvarov** [comment]
Closing, as issue was fixed with the PR #7742
